src/components/contact-govuk-one-login/contact-govuk-one-login-controller.ts

```diff
@@ -12,18 +12,51 @@ export function contactGet(req: Request, res: Response): void {
       "Request to contact-govuk-one-login page did not contain a valid fromURL in the request or session"
     );
   }
+  // optional fields from mobile
+  const themeId = getValueFromRequestOrSession(req, "themeId");
+  const appSessionId = getValueFromRequestOrSession(req, "appSessionId");
+  const appErrorCode = getValueFromRequestOrSession(req, "appErrorCode");
+
   const data = {
     fromURL: fromURL,
+    appSessionId: appSessionId,
+    appErrorCode: appErrorCode,
+    themeId: themeId,
   };
   res.render(CONTACT_ONE_LOGIN_TEMPLATE, data);
 }
 
 const getFromUrlAndSaveIt = (request: Request): string => {
   const fromURLFromRequest = request.query.fromURL as string;
-  if (isValidUrl(fromURLFromRequest)) {
-    request.session.fromURL = fromURLFromRequest;
-    return fromURLFromRequest;
+  if (fromURLFromRequest) {
```

Comment:
I think we should try avoiding nest if statements, this can be refactored as
`
if (fromURLFromRequest && isValidUrl(fromURLFromRequest)) {
    request.session.fromURL = fromURLFromRequest;
    return fromURLFromRequest;
} else if (fromURLFromRequest) {
    logger.error("fromURL in request query for contact-govuk-one-login page did not pass validation");
}
`

Reply:
done